src/server/clientiface.cpp

```diff
@@ -435,32 +418,33 @@ void RemoteClient::GotBlock(v3s16 p)
 
 void RemoteClient::SentBlock(v3s16 p)
 {
-	if (m_blocks_sending.find(p) == m_blocks_sending.end())
-		m_blocks_sending[p] = 0.0f;
-	else
+	if (!m_blocks_sending.insert(p).second)
 		infostream<<"RemoteClient::SentBlock(): Sent block"
 				" already in m_blocks_sending"<<std::endl;
 }
 
-void RemoteClient::SetBlockNotSent(v3s16 p)
+void RemoteClient::SetBlockNotSent(v3s16 p, bool low_priority)
 {
-	m_nothing_to_send_pause_timer = 0;
-
 	// remove the block from sending and sent sets,
-	// and mark as modified if found
-	if (m_blocks_sending.erase(p) + m_blocks_sent.erase(p) > 0)
-		m_blocks_modified.insert(p);
+	// and reset the scan loop if found
+	if (m_blocks_sending.erase(p) + m_blocks_sent.erase(p) > 0) {
+		m_nothing_to_send_pause_timer = 0;
+		if (!m_perf_tradeoffs && !low_priority) {
```

Comment:
What I think would be nice here is a short explanation on what happens if `low_priority == true`.
The send loop will start again but might only get to the updated block on the next iteration? Is that it?